File: android/app/src/main/java/fr/xebia/magritte/classifier/ClassifierContract.java
package fr.xebia.magritte.classifier;

import android.content.Context;
import android.graphics.Bitmap;

import java.util.List;
import java.util.Locale;

import fr.xebia.magritte.model.FruitResource;

public interface ClassifierContract {

    interface View {

        Context getContext();

        void displayRecognitions(List<Classifier.Recognition> recognitionList);

        void displayTopMatch(FruitResource fruit);

        void speakResult(String title);
    }

    interface Presenter {

        void recognizeImage(Bitmap bitmap);

        String getLocalizedString(Locale locale, int resId);
    }
}

File: android/app/src/main/java/fr/xebia/magritte/classifier/ClassifierContract.kt
package fr.xebia.magritte.classifier

import android.content.Context
import android.graphics.Bitmap
import fr.xebia.magritte.model.FruitResource
import java.util.*

interface ClassifierContract {

    interface View {

        val context: Context

        fun displayRecognitions(recognitionList: MutableList<Classifier.Recognition>)

        fun displayTopMatch(fruit: FruitResource)

        fun speakResult(title: String)
    }

    interface Presenter {

        fun recognizeImage(bitmap: Bitmap)

        fun getLocalizedString(locale: Locale, resId: Int): String
    }
}
